platform/darwin/src/MGLStyle.mm

```diff
@@ -221,7 +221,7 @@ - (MGLSource *)sourceWithIdentifier:(NSString *)identifier
 }
 
 - (MGLSource *)sourceFromMBGLSource:(mbgl::style::Source *)rawSource {
-    if (MGLSource *source = rawSource->peer.empty() ? nil : mbgl::any_cast<SourceWrapper>(rawSource->peer).source) {
+    if (MGLSource *source = rawSource->peer.has_value() ?  mbgl::util::any_cast<SourceWrapper>(&(rawSource->peer))->source : nil) {
```

Comment:
Can this continue to use the throwing version of `any_cast` rather than the null-returning version, so that if it fails it's not undefined behavior?